lib/AST/ASTPrinter.cpp

```diff
@@ -2406,21 +2415,35 @@ void PrintAST::visitVarDecl(VarDecl *decl) {
       Printer << " ";
     }
   }
-  printContextIfNeeded(decl);
-  recordDeclLoc(decl,
-    [&]{
-      Printer.printName(decl->getName());
-    });
-  if (decl->hasInterfaceType()) {
-    Printer << ": ";
-    auto tyLoc = decl->getTypeLoc();
-    if (!tyLoc.getTypeRepr())
-      tyLoc = TypeLoc::withoutLoc(decl->getInterfaceType());
+
+  {
+    llvm::SaveAndRestore<bool>
+      printAsIUO(Options.PrintOptionalAsImplicitlyUnwrapped);
 
     if (decl->getAttrs().hasAttribute<ImplicitlyUnwrappedOptionalAttr>())
-      printTypeLocForImplicitlyUnwrappedOptional(tyLoc);
-    else
+      Options.PrintOptionalAsImplicitlyUnwrapped = true;
+
+    auto pbd = decl->getParentPatternBinding();
+    if (pbd && !Options.ExplodePatternBindingDecls) {
+      auto &entry = pbd->getPatternEntryForVarDecl(decl);
+      printPatternBindingEntry(entry);
+      printAccessors(decl);
```

Comment:
This seems like it'll print the same entry twice when there are two VarDecls in the same entry.

Reply:
It might just be more profitable and cleaner to figure out when we print a single VarDecl without going through the pattern binding. That's what's really happening here.